pkg/cloud/generic/actuators/machine/actuator.go

```diff
@@ -52,32 +52,72 @@ func NewActuator(params ActuatorParams) (*Actuator, error) {
 
 // Create creates a machine and is invoked by the Machine Controller
 func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	// Create is only called if a Machine resource exists. It is a
+	// configuration error if a Cluster does not exist in the same
+	// namespace as the Machine. By returning an error here we ensure
+	// the resource will be requeued for processing later.
+	if cluster == nil {
+		return fmt.Errorf("machine %s does not have cluster in namespace %s", machine.Name, machine.Namespace)
+	}
+
 	log.Printf("Creating machine %v for cluster %v.", machine.Name, cluster.Name)
-	return fmt.Errorf("TODO: Not yet implemented")
+
+	providerID := fmt.Sprintf("%s-%s", cluster.Name, machine.Name)
+	machine.Spec.ProviderID = &providerID
+	machine, err := a.machinesGetter.Machines(machine.Namespace).Update(machine)
+	if err != nil {
+		log.Printf("Failed to set Machine.Spec.ProviderID %s: %v", providerID, err)
+		return fmt.Errorf("failed to set Machine.Spec.ProviderID %s: %v", providerID, err)
+	}
+
+	// TODO: Call webhook to allocate infrastruction
+
+	return nil
 }
 
 // Delete deletes a machine and is invoked by the Machine Controller
 func (a *Actuator) Delete(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	if cluster == nil {
+		return fmt.Errorf("machine %s does not have cluster in namespace %s", machine.Name, machine.Namespace)
+	}
+
 	log.Printf("Deleting machine %v for cluster %v.", machine.Name, cluster.Name)
-	return fmt.Errorf("TODO: Not yet implemented")
+
+	if machine.Spec.ProviderID == nil {
+		log.Printf("Machine %s-%s does not have ProviderID so there is nothing to delete", cluster.Name, machine.Name)
+		return nil
+	}
+
+	// TODO: Call webhook to release infrastructure
+
+	machine.Spec.ProviderID = nil
+	machine, err := a.machinesGetter.Machines(machine.Namespace).Update(machine)
```

Comment:
This shouldn't be necessary here, since we are in the delete workflow, once this returns a non-error, the finalizer will be removed and the Machine object will disappear.